docs/proposals/20250325-distributed-inference.md

```diff
@@ -133,38 +132,125 @@ env:
+- **Worker Rejoining Delay:** In the case of a leader restart, existing worker pods may take 1–2 minutes to recognize a new leader, as they may continue sending heartbeats to the old leader.
+- **Ray Actor Health & Count Validation**: The Ray cluster’s health depends on having enough alive actors matching the world size. Dead or missing actors must be detected.
+- **Failure Intolerance**: The current design does not tolerate worker or leader pod failures gracefully. Although leader restart does not require all workers to restart, each worker restart requires a leader restart for synchronization purposes, and sequential node upgrades may incur significant downtime.
+
+To address these issues, the following probing strategy is proposed:
+
+|            | Liveness Probe (Triggers Container Restart on Failure)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    | Readiness Probe (Does Not Trigger Container Restart on Failure)                         |
+| ---------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | --------------------------------------------------------------------------------------- |
+| **Leader** | Checks Ray cluster health via `python3 -c 'import ray; ray.init(); from ray._private.state import actors as get_all_actors; get_all_actors()'`, filtering by the correct job id to exclude duplicates. Ensures zero dead actors and alive actors equal StatefulSet replica count. Excludes `/health` endpoint.                                                                                                                                                                                                                                                                                                            | Checks `/health` endpoint at `$VLLM_LEADER_SVC:5000` after some initial delay.          |
```

Comment:
please update the health check script here as we discussed offline. We don't need to write the exact code here.
Just say we list actor through dashboard API.